File: Chapter9ActivityHook1/app/src/main/java/com/mars/activityhook1/HostApplication.java
package com.mars.activityhook1;

import android.app.Application;
import android.content.Context;

/**
 * Created by geyan on 2020/10/27
 */
public class HostApplication extends Application {

    private static Context mContext;

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        mContext = base;
    }

    public static Context getContext() {
        return mContext;
    }
}

File: Chapter9ActivityHook1_V2/app/src/main/java/com/mars/chapter9activityhook1_v2/HostApplication.kt
package com.mars.chapter9activityhook1_v2

import android.app.Application
import android.content.Context
import com.mars.chapter9activityhook1_v2.hook.AMSHookHelper
import com.mars.chapter9activityhook1_v2.hook.LoadedApkClassLoaderHookHelper
import com.mars.components.util.file.FileCopyUtils

/**
 * Created by geyan on 2021/1/16
 */
class HostApplication : Application() {

    companion object {
        private var mContext: Context? = null

        @JvmStatic
        fun getContext(): Context? {
            return mContext
        }
    }

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        mContext = base
        FileCopyUtils.extractAssets(base, "plugin.apk")
        LoadedApkClassLoaderHookHelper.hookLoadedApkInActivityThread(getFileStreamPath("plugin.apk"))
        AMSHookHelper.hookActivityManagerNative()
        AMSHookHelper.hookActivityThread()
    }

}